### ISSUE TITLE
dot method not defined

### ISSUE THREAD

**Issue by johnnychen94** [JuliaGraphics/ColorVectorSpace.jl#101]
```julia
julia> using ImageCore

julia> using LinearAlgebra

julia> x = rand(Gray, 3, 3)
3×3 Array{Gray{Float64},2} with eltype Gray{Float64}:
 Gray{Float64}(0.54763)   Gray{Float64}(0.684368)  Gray{Float64}(0.213733)
 Gray{Float64}(0.703994)  Gray{Float64}(0.316331)  Gray{Float64}(0.604376)
 Gray{Float64}(0.970118)  Gray{Float64}(0.970798)  Gray{Float64}(0.509367)

julia> dot(channelview(x), channelview(x))
3.9179152985873418

julia> dot(x, x)
ERROR: MethodError: no method matching iterate(::Gray{Float64})
Closest candidates are:
  iterate(::Core.SimpleVector) at essentials.jl:568
  iterate(::Core.SimpleVector, ::Any) at essentials.jl:568
  iterate(::ExponentialBackOff) at error.jl:199
  ...
Stacktrace:
 [1] dot(::Gray{Float64}, ::Gray{Float64}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/LinearAlgebra/src/generic.jl:673
 [2] dot(::Array{Gray{Float64},2}, ::Array{Gray{Float64},2}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.1/LinearAlgebra/src/generic.jl:728
 [3] top-level scope at none:0
```

**Comment by timholy** [comment]
I would say the missing operation is `dot(::Gray, ::Gray)`, not `iterate`.

**Comment by johnnychen94** [comment]
oh yes, it's fixed in #102
